arbitrator/jit/src/syscall.rs

```diff
@@ -557,7 +554,8 @@ pub fn js_copy_bytes_to_js(env: &WasmEnvArc, sp: u32) {
                     let len = std::cmp::min(src_len, dest_len) as usize;
 
                     // Slightly inefficient as this allocates a new temporary buffer
-                    buf[..len].copy_from_slice(&sp.read_slice(src_ptr, len as u64));
+                    let slice = sp.read_slice(src_ptr, len as u64);
+                    buf[..len].copy_from_slice(&slice);
```

Comment:
Nitpick but `slice` is actually a vec not a slice. Maybe call it `data`?

Reply:
If we really wanted to be efficient here we could have a `read_slice` method that directly copies into `buf`, but that's probably not necessary

Reply:
Oops